cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesJarUtil.java

```diff
@@ -0,0 +1,238 @@
+package com.devonfw.cobigen.impl.util;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.devonfw.cobigen.api.constants.TemplatesJarConstants;
+
+/**
+ *
+ */
+public class TemplatesJarUtil {
+
+    /**
+     * Filters the files on a directory so that we can check whether the templates jar are already downloaded
+     */
+    static FilenameFilter fileNameFilterJar = new FilenameFilter() {
+
+        @Override
+        public boolean accept(File dir, String name) {
+            String lowercaseName = name.toLowerCase();
+            String regex = TemplatesJarConstants.JAR_FILE_REGEX_NAME;
+
+            Pattern p = Pattern.compile(regex);
+            Matcher m = p.matcher(lowercaseName);
+            if (m.find()) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    };
+
+    /**
+     * Filters the files on a directory so that we can check whether the templates jar are already downloaded
+     */
+    static FilenameFilter fileNameFilterSources = new FilenameFilter() {
+
+        @Override
+        public boolean accept(File dir, String name) {
+            String lowercaseName = name.toLowerCase();
+            String regex = TemplatesJarConstants.SOURCES_FILE_REGEX_NAME;
+
+            Pattern p = Pattern.compile(regex);
+            Matcher m = p.matcher(lowercaseName);
+            if (m.find()) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    };
+
+    /**
+     * @param isDownloadSource
+     *            true if downloading source jar file
+     * @param templatesDirectory
+     *            directory where the templates jar are located
+     * @return fileName Name of the file downloaded
+     * @throws MalformedURLException
+     *             {@link MalformedURLException} occurred
+     * @throws IOException
+     *             {@link IOException} occurred
+     */
+    public static String downloadJar(boolean isDownloadSource, File templatesDirectory)
+        throws MalformedURLException, IOException {
+        String mavenUrl =
```

Comment:
maybe we should at least make this groupId and artifactId and version also configurable, meaning, we have a second function providing the ability to download a different jar and renaming this function to downloadLatestDevon4jTemplates.